plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts

```diff
@@ -77,8 +77,10 @@ export class AwsLambdaInstrumentation extends InstrumentationBase {
   private _traceForceFlusher?: () => Promise<void>;
   private _metricForceFlusher?: () => Promise<void>;
 
-  constructor(protected override _config: AwsLambdaInstrumentationConfig = {}) {
-    super('@opentelemetry/instrumentation-aws-lambda', VERSION, _config);
+  protected override _config!: AwsLambdaInstrumentationConfig;
```

Comment:
This is another bit we may want to align, which is the way we type config in our instrumentations. Here we use the `override` keyword so we tell typescript to treat the property as an `AwsLambdaInstrumentationConfig` type. And in other instrumentations we have a getter that casts the config property to the type we need


```ts
getConfig(): AwsLambdaInstrumentationConfig {
  return this._config as AwsLambdaInstrumentationConfig;
}
```

Should we encourage one over he other?

Reply:
Once https://github.com/open-telemetry/opentelemetry-js/pull/4659 lands, we can cleanup all the config handling variations as the base class would already be typed correctly. Eager to apply this cleanup in all contrib instrumentations